source/portable/NetworkInterface/loopback/loopbackNetworkInterface.c

```diff
@@ -0,0 +1,184 @@
+ * The function FreeRTOS_IPInit() will call it to initialice the interface and end-point
+ * objects.  See the description in FreeRTOS_Routing.h. */
+    NetworkInterface_t * pxFillInterfaceDescriptor( BaseType_t xEMACIndex,
+                                                    NetworkInterface_t * pxInterface )
+    {
+        pxLoopback_FillInterfaceDescriptor( xEMACIndex, pxInterface );
+    }
+
+#endif /* ( ipconfigCOMPATIBLE_WITH_SINGLE != 0 ) */
+/*-----------------------------------------------------------*/
+
+NetworkInterface_t * pxLoopback_FillInterfaceDescriptor( BaseType_t xEMACIndex,
+                                                         NetworkInterface_t * pxInterface )
+{
+/* This function pxLoopback_FillInterfaceDescriptor() adds a network-interface.
+ * Make sure that the object pointed to by 'pxInterface'
+ * is declared static or global, and that it will remain to exist. */
+
+    memset( pxInterface, '\0', sizeof( *pxInterface ) );
+    pxInterface->pcName = "Loopback";                /* Just for logging, debugging. */
+    pxInterface->pvArgument = ( void * ) xEMACIndex; /* Has only meaning for the driver functions. */
+    pxInterface->pfInitialise = prvLoopback_Initialise;
+    pxInterface->pfOutput = prvLoopback_Output;
+    pxInterface->pfGetPhyLinkStatus = prvLoopback_GetPhyLinkStatus;
+
+    FreeRTOS_AddNetworkInterface( pxInterface );
+    xLoopbackInterface = pxInterface;
+
+    return pxInterface;
+}
+/*-----------------------------------------------------------*/
+
+static BaseType_t prvLoopback_GetPhyLinkStatus( NetworkInterface_t * pxInterface )
+{
+    /* This function returns true if the Link Status in the PHY is high. */
+    ( void ) pxInterface;
+    return pdTRUE;
+}
+/*-----------------------------------------------------------*/
+
+static BaseType_t prvLoopback_Output( NetworkInterface_t * pxInterface,
+                                      NetworkBufferDescriptor_t * const pxGivenDescriptor,
+                                      BaseType_t bReleaseAfterSend )
+{
+    NetworkBufferDescriptor_t * pxDescriptor = pxGivenDescriptor;
+
+    ( void ) pxInterface;
+
+    IPPacket_t * a = ( IPPacket_t * ) ( pxDescriptor->pucEthernetBuffer );
+
+    if( a->xEthernetHeader.usFrameType == ipIPv4_FRAME_TYPE )
+    {
+        usGenerateProtocolChecksum( pxDescriptor->pucEthernetBuffer, pxDescriptor->xDataLength, pdTRUE );
+    }
+
+    {
+        MACAddress_t xMACAddress;
+
+        if( pxDescriptor->pxEndPoint->bits.bIPv6 != 0 )
+        {
+            #if ( ipconfigUSE_IPv6 != 0 )
+                if( xIsIPv6Loopback( &( pxDescriptor->xIPAddress ) ) != pdFALSE )
+                {
+                    vNDRefreshCacheEntry( &xMACAddress, &( pxDescriptor->xIPAddress.xIP_IPv6 ), pxDescriptor->pxEndPoint );
+                }
+            #endif
+        }
+        else
+        {
+            #if ( ipconfigUSE_IPv4 != 0 )
+                if( xIsIPv4Loopback( pxDescriptor->xIPAddress.ulIP_IPv4 ) )
```

Comment:
I couldn't find a definition for `xIsIPv4Loopback`, as we didn't had that function defined in the stack is it missed?

Reply:
Marking this as resolved as #1020 adds the mentioned change.